app/controllers/v0/sign_in_controller.rb

```diff
@@ -225,6 +228,46 @@ def introspect
       render json: { errors: e }, status: :unauthorized
     end
 
+    def fetch_openid_configuration
+      render json: openid_configuration_data
```

Comment:
Controllers need to be 'thin', this means that code that can be defined in a separate class, should be defined in a separate class. So you'll need to create a service object that this controller calls and gets the openid data from